certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java

```diff
@@ -75,22 +73,30 @@ public class VCIssuanceServiceImpl implements VCIssuanceService {
 
     @Override
     public CredentialResponse getCredential(CredentialRequest credentialRequest) {
+        boolean isValidCredentialRequest = new CredentialRequestValidatorFactory().isValid(credentialRequest);
+        if(!isValidCredentialRequest) {
+            throw new InvalidRequestException(ErrorConstants.INVALID_REQUEST);
+        }
+        System.out.println("getCredential parsedAccessToken "+parsedAccessToken);
+
         if(!parsedAccessToken.isActive())
             throw new NotAuthenticatedException();
 
         String scopeClaim = (String) parsedAccessToken.getClaims().getOrDefault("scope", "");
         CredentialMetadata credentialMetadata = null;
-        for(String scope : scopeClaim.split(Constants.SPACE)) {
-            Optional<CredentialMetadata> result = getScopeCredentialMapping(scope);
-            if(result.isPresent()) {
-                credentialMetadata = result.get(); //considering only first credential scope
-                break;
+        if(!credentialRequest.getFormat().equals(VCFormats.MSO_MDOC)){
```

Comment:
TODO(vharsh): discuss with @jainhitesh9998 why the scope validation not required for mDoc VCs.

Reply:
Scope validation is now added for mdoc VCs as well